pennylane/optimize/gradient_descent.py

```diff
@@ -127,6 +127,9 @@ def compute_grad(objective_fn, args, kwargs, grad_fn=None):
         grad = g(*args, **kwargs)
         forward = getattr(g, "forward", None)
 
+        if len(args) == 1:
+            grad = (grad,)
+
```

Comment:
We have to turn grad into a tuple because we're reversing what we did to argnum earlier, right?

Reply:
Yep, the optimizer logic currently assumes `grad` is always iterable with respect to argument number.

I figured, easier to leave this logic intact and simply re-tuple the grad

Reply:
Agreed.